### ISSUE TITLE
The BIG Grants Round 7 Feedback Thread

### ISSUE THREAD

**Issue by owocki** [gitcoinco/web#7408]
Welcome to the BIG Gitcoin Grants CLR Round 7 Product Feedback Thread!

Grants is one of our core products alongside [Gitcoin Hackathons](https://gitcoin.co/hackathon-list). We improve grants on an iterative cycle around Gitcoin Grants rounds. 

The next big effort to improve grants will be CLR round 8 (Dec 2020)

This is a thread where you can catalogue product feedback for us to work on for the next round.  

TLDR:
- Love Round 7? Post below
- Hate something about Round 7? Post below
- Found a bug? Post below
- Got a comment? Post below

**Comment by peekavr** [comment]
My grant doesn't currently reflect any matching. Wondering if there's something I forgot to do. Thanks for all your work!

**Comment by codyborn** [comment]
My donation transaction succeeded but the confirmation wasn't successful.
https://etherscan.io/tx/0xfbe9692ce13c00dd63a13b0dd880a38bf788f1b08439e14ac4a892bf4bbf6570
<img width="392" alt="image" src="https://user-images.githubusercontent.com/11261382/93287892-83b0af80-f78f-11ea-8e58-1567edc80189.png">

**Comment by owocki** [comment]
@mds1 i think u are working on a similar issue at https://github.com/gitcoinco/web/issues/7421 right

**Comment by mds1** [comment]
> My donation transaction succeeded but the confirmation wasn't successful.

@codyborn Don't worry, all of your zkSync transactions were also successful! There's a simple bug that causes this at the end that we need to fix, but no action is needed by you 🙂

**Comment by niran** [comment]
One of our donors is saying that a bug won't let them donate. "Invalid json rpc response." They tried on a few different browsers, but had the same problem. Does this sound familiar?

**Comment by mds1** [comment]
Hey @niran!

One reason may be that the zkSync servers were down at that time. Have they tried again since? If it's still providing issues, can you/they provide more details about what was in their cart (mainly the list of tokens used), when exactly this happened, and any relevant tx hashes? If they're technically savvy, you can also dig into the "Network" tab in the browser console, find the specific request that failed, and pull out a detailed response from the request

Feel free to DM me the info on telegram if you don't want to post it publicly here!

**Comment by niran** [comment]
Still working on getting more information about the last issue, but here's a separate one: the cart screen silently fails to load if I haven't connected MetaMask yet.

![image](https://user-images.githubusercontent.com/106437/93502972-16ab3000-f8dd-11ea-8979-4a7e6f03b66c.png)

**Comment by mds1** [comment]
@niran Ah yes that one is a known issue! In the meantime just make sure MetaMask is unlocked/connected for now to work around that

**Comment by owocki** [comment]
suggestion to split community rounds by region

https://bits.owocki.com/eDuBGR95

**Comment by owocki** [comment]
What about a social recommender à la 'enter your friends gitcoin handle' or 'look up your friends in this list of everyone who funded grants'?
https://twitter.com/chriscyber2000/status/1307321605094285315

**Comment by owocki** [comment]
Did a few gitcoin grant contributions: https://twitter.com/ksaitor/status/1308225557406990338
Some unsolicited feedback, if I may :)
The link and the page that's generated and shared on twitter can be improved imho.
I thought that the linked page (https://gitcoin.co/grants/cart/bulk-add/329;5;283,821;5;283,238;5;283,40;5;283,:ksaitor) would show some sort of page with user's contributions, but instead it right away adds these contributions to cart. Which felt a bit "pushy" perhaps?
Also it right away rewrote the url to https://gitcoin.co/grants/cart - which also felt confusing for a bit.
Aaaand the page doesn't seem to open when the user is not logged in. Why might not be good, if you are trying to convert people who haven't heard about gitcoin yet. 
My 2 cents. Something to pass to your growth team.
Warm regards,
Raman

**Comment by owocki** [comment]
video intros https://twitter.com/Anne_Connelly/status/1308780464136749066

**Comment by owocki** [comment]
https://twitter.com/owocki/status/1308787414727499776

**Comment by santteegt** [comment]
UI needs to improve the gas fees estimate when using zkSync. I used 63+66 gWei as gas cost (approval + Confirm) and I ended up spending > $4 in tx fees instead of ~ $1.615 estimated. Now I ended up in doubt of what would be cheaper given the total amount of grants donations using a single token (DAI), doing an L2 vs L1

![image](https://user-images.githubusercontent.com/926341/94086195-db729a80-fdcf-11ea-9fb5-446ddaf67fd4.png)

Txs on Etherscan:
* https://etherscan.io/tx/0x0b2f735ab88bc9eb72c67463ea6de68b918a8270306cf85890c2fcfec94dff5d
* https://etherscan.io/tx/0xc352c0553130c4e63f57290fb3e706a6d2873de61bf12cb6223aa04bf43eb1af 

Would be cool to have something better than estimated % of gas savings. i.e. showing a more accurate gas estimate for both choices L2 vs L1

**Comment by mds1** [comment]
Hey @santteegt, thanks for the feedback. Agreed it's a bit confusing. Getting a very accurate cost estimate can be tricky, so we're always working on better ways to estimate and display the fees.

To clarify how fees work, there's two sets of fees:
- The fees paid from your regular web3 wallet to deposit funds into zkSync. This is what you see when you confirm the transaction on the final step for ~$4.
- Then there's fees to send each transfer on zkSync, which is the 1.615 DAI you see in the modal. The 1.615 DAI estimate is very conservative, and in your case the total fees were about 0.6 DAI. So we withdrew 7 DAI plus the fees (since it withdrew 8.724 DAI instead of 8.615, we likely have a small bug in the fee estimate there), then send your transfers on zkSync. Any leftover funds not used for fees are in your zkSync wallet on L2, which you can view at https://wallet.zksync.io/ or https://zkscan.io/

Given that it seems you had 6-7 items in your cart (from looking at zkScan transfers), zkSync was the cheaper option in your case 🙂

**Comment by owocki** [comment]
fernando mentioned that he wants the abilityt o upload a phooto; not make an avatar.

**Comment by owocki** [comment]
default token amount https://twitter.com/thegrifft/status/1309223712223232003

**Comment by owocki** [comment]
https://twitter.com/stellarmagnet/status/1310711507660025862

suggestions from yalda + niran

```
Maybe add another box at checkout for people to opt into matching for their dev fund contribution? Like the box on our tax returns for the election campaign fund that doesn't cost you anything when you check it https://en.wikipedia.org/wiki/Presidential_election_campaign_fund_checkoff?wprov=sfla1


I think adding this opt-in would make sense. Slight platform advantage, but at least the user takes the action to recognize it.

```

**Comment by owocki** [comment]
https://twitter.com/impe83/status/1310806063139094531?s=21

what can we do about scam grants?

**Comment by owocki** [comment]
showcase share images https://bits.owocki.com/P8uyjEOO

**Comment by owocki** [comment]
also: Hey this would be cool to have, a template to thank funders and tweet them, any idea on how we can hack it?
https://twitter.com/coinosis/status/1311077836690780161

**Comment by owocki** [comment]
https://twitter.com/defi_latam/status/1311102408605532162?s=12

if you update a grant activity; it should set the idle to 0

**Comment by owocki** [comment]
feedback thread on twitter https://twitter.com/owocki/status/1311295170034319362

**Comment by willsputra** [comment]
The only thing I was confused by is that there were 3 buttons. I did the first one and the 2nd one then said I didn't have to do it (I think the 2nd step was approve ERC20s?). After I did that at the top it said "Please don't close this window, estimated wait is 2 - 4 minutes)", so I left it. I came back a half hour later and it was in the same state. That's when I realized there was the 3rd step I had ignored, pressed that button then everything went through and made sense
-> make the relation between wait times and steps clearer (which step does this "2-4 minutes" correspond to?)

https://twitter.com/coinyon/status/1308009037779398656
-> provide option if people want to deposit funds to zksync wallet directly (not through gitcoin)

**Comment by owocki** [comment]
https://twitter.com/letsgobankless/status/1311329721473523714

show total clr match on checkout (and in emails)

**Comment by owocki** [comment]
i got asked why the brightid verification hadn't run yet for a user

```
@Owocki how can I be sure I'm brightID verified?
i have the check next to my name, have gone to a verfication party, connected with a bunch of peopl
and connected it to gitcoin, but now it's prompting me to go to another verification party
```
the answer was

```
@Trent the brightid<>gitcoin verificaiton sync script runs once per day; and looks like it ahdnt run yet today.  im going to make a feedback note that it should run more often and/or user should be able to manually trigger it
i just re-ran it for you + your profile is def verified us
```

we should run it as soon as the user verifies i think, or at least 1x a day

**Comment by owocki** [comment]
add analytics to collections. view counts, add to cart counts, etc. displpay them publicly

**Comment by owocki** [comment]
Trends feedback

```
the "apply to all" button is a little too easy to engage
messed up all my totals a few times
the profile size is an improvement over the boxes of previous rounds, but i wish they were even smaller!
give me a spreadsheet
```

```
Watch for some grants that don't accept dai. I had a weird bug with Austin's grant.
```

spammy dupe grants https://bits.owocki.com/p9uwxmXL

HTML spaghetti shows up on gitcoinlouis grant https://bits.owocki.com/qGuvpL7v

"accidentally clicking a grant instead of opening a new tab and then having to scroll back down" is annoying

**Comment by owocki** [comment]
https://bits.owocki.com/o0umpzwn
https://bits.owocki.com/KouwD2o4
https://bits.owocki.com/9ZuB2nZO

hey @apbendi @mds1  one potential showstopper; from a user

```
I commented "u up" on sam sun, not the defiant haha
and the daddy comment was for DeFi_Dad, not lighthouse :sweat_smile:
```

do you know why the comments input in the cart were given to the wrong grants?  is there maybe some sort of 'off by one' error somewhere?

**Comment by mds1** [comment]
Hey @tvanepps! Did you remove any grants from your cart? I'm guessing that may be the cause, and for some reason the comments didn't shift accordingly. It's possible this is fixed now after #7573, but I'll need to look into this to confirm

**Comment by mds1** [comment]
Send emails later in the checkout process. They are sent immediately, which is confusing for zkSync checkout because the user thinks they can leave the page after they receive those emails, but in reality they must wait until everything is complete

**Comment by owocki** [comment]
automate the cart_reminder emails in the final few hours/days of the round

sent_cart_reminder

**Comment by owocki** [comment]
TODO: in the txn validator, do stuff related to long running pending txns - we should notify users of these + have them do something

**Comment by cryptoouf-blog** [comment]
Loved it. You are doing an amazing job.

I really liked the use zkSync  for donation, it's easy and quick.
 But i didn't really understand how it works. You just have to sign 3 times. But can you also add a pop up window or something to show what are the fees for the transaction. Maybe it's only me, but at the end i was a bit lost regarding what i payed ...

**Comment by owocki** [comment]
emails should no longer say things like "1 time" as the product no longer supports multiple contributions / subscriptoins https://bits.owocki.com/qGuvymkK

**Comment by owocki** [comment]
on transactions page, add pagination https://bits.owocki.com/ApuLnpbp
and csv export

**Comment by tjayrush** [comment]
I'd like to see an alternative to the 'infinite scroll'. Infinite scroll makes it very difficult to consume the whole grants database in one shot. In fact, the only thing infinite about the scroll is how long it takes to see the data.

May I suggest a paginated table with configurable columns and robust sorting and filtering? It's pretty easy to add if you're a React app (are you? not sure?). This would allow anyone to find anything that they consider important. For example, I would exclude any grant that has 'dashboard' in it and most that have DeFi and all of the top 30 (as they've already gotten their share).

If you add 'checkboxes' and inline forms to adjust the payment amounts, that would make the process even that much more fluid.

Love your work. Keep up the excellent mission.

**Comment by owocki** [comment]
@frankchen07 suggests a 'similarity index' for helping catch collusion

**Comment by octavioamu** [comment]
"People who contributed to this grant also contributed to > "

**Comment by owocki** [comment]
* put a clear path to the matching grants when live on the homepage. Harder to find than it should be

* better descriptions of the applicants on the page. Give them more direction on how to write. “your ask in 140 characters”

* delegate money to someone to make grants for me.

* Donate without creating account

* got a prompt in metamask saying gas would cost .74 eth. Went and approved and it was actually . Ended up being .18 in gas

ideas from ron ^^

**Comment by cryptoouf-blog** [comment]
> * Donate without creating account

There is this tool, that is amazing (they had a grant) , you can donate directly on twitter, and if i'm not wrong you don't need an account on github (maybe double check that)
https://mask.io/

**Comment by mds1** [comment]
- UI can still be more clear about zkSync tx fees
- Improve zkSync fee estimates based on whether recipient has used zkSync (currently we assume all transfers are transfers to new addresses, since that is more costly, but this is a very conservative assumption)

**Comment by owocki** [comment]
https://www.youtube.com/watch?v=mUZOHZFZqGM

we could perhaps add collabland integration

**Comment by owocki** [comment]
v: It's actually not clear to me that adding collections to cart is a behaviot worth encouraging

**Comment by owocki** [comment]
KYC feedback https://twitter.com/gitcoin/status/1313838085633835009

**Comment by owocki** [comment]
Grants 
- Directly say thank you to public goods funders 
- public goods justice league art bounty .. fighting moloch

**Comment by owocki** [comment]
people reach out to me at the start of the round to ask how they can maximize their grant, and we dont have a "how to promtoe your grant" guide to give them. as a result, people who have played the game the longest + have insider knowledge rise to the top.

**Comment by owocki** [comment]
partnership with yearn/one click portal thing

**Comment by owocki** [comment]
grant detail page needs cleanup

**Comment by owocki** [comment]
https://twitter.com/lefterisjp/status/1318320082980921351?s=21 distributed curation idea

**Comment by owocki** [comment]
https://github.com/gitcoinco/web/issues/7726

**Comment by frankchen07** [comment]
instead of infinite scroll, a better system for looking at a smaller subset of grants. similar to collections, maybe just surfacing collections in a more ecommerce sort of way

**Comment by owocki** [comment]
interesting sybil technique https://twitter.com/financedotvote/status/1327361133695672320

**Comment by frankchen07** [comment]
^ oh that is interesting. we can do a NFTs already, and then we can link to the GC token when released?

**Comment by owocki** [comment]
similar feedback thread, but for round 8 => https://github.com/gitcoinco/web/issues/8000
